packages/toolpad-app/src/components/AppEditor/ConnectionEditor/index.tsx

```diff
@@ -64,20 +65,26 @@ function ConnectionEditorContent<P>({
 
   const dataSourceType = connectionNode.attributes.dataSource.value;
   const dataSource = dataSources[dataSourceType];
+  const connectionEditorContext = React.useMemo(
+    () => ({ appId, connectionId: connectionNode.id }),
+    [appId, connectionNode.id],
+  );
 
   return (
     <Box className={className} sx={{ width: '100%', height: '100%', p: 3 }}>
       <Stack spacing={1}>
         <NodeNameEditor node={connectionNode} />
         {dataSource ? (
-          <ConnectionParamsEditor
-            dataSource={dataSource}
-            value={connectionNode.attributes.params.value}
-            onChange={handleConnectionChange}
-            handlerBasePath={`/api/dataSources/${dataSourceType}`}
-            appId={appId}
-            connectionId={connectionNode.id}
-          />
+          <QueryEditorContextProvider value={connectionEditorContext}>
```

Comment:
Wouldn't it be confusing to use `QueryEditorContext` in the `ConnectionEditor`? Do we need a `ConnectionEditorContext`?

Reply:
I thought of adding that but it felt very redundant because we need the exact same attributes; in that case we would also need to create a separate version of the `usePrivateQuery` hook

Reply:
do we need to rename it to `DataSourceConnectionContext`?

Reply:
The `DataSource` part seems redundant to me - `ConnectionContext` feels succinct enough. What do you think?